odoo/addons/test_pylint/tests/test_pylint.py

```diff
@@ -1,24 +1,47 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import contextlib
 import logging
 try:
     import pylint
+    from pylint import lint
+    from pylint.reporters import BaseReporter
 except ImportError:
     pylint = None
-import subprocess
+import sys
 from distutils.version import LooseVersion
-from os import devnull
 from os.path import join
 
 from odoo.tests.common import TransactionCase
 from odoo import tools
 from odoo.modules import get_modules, get_module_path
 
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
```

Comment:
Is there a reason for not using io everywhere? Also beware, io.BytesIO and io.StringIO are *much* stricter about their values than the old StringIO.StringIO is, even in Python 2 giving bytes to io.StringIO or text to io.BytesIO will blow up.